### PR TITLE
Notifications Overhaul #1158, Enhancement #1154, and more....

### PR DESCRIPTION
**Pull request by devEricA**
This is a series of changes that I made to Defectdojo's Notifications as a response to Issue #1158 

The following notifications were changed:
- Review Requested : Now displays the requested users that are intended to review a finding.
- Engagement Added: Now displays the engagement's name instead of the engagement's product.
- Import Scan Results: Now Displays the scan that was uploaded, used to only display "More information can be found here: ....."
- Test Added: Now displays the test's title along with the test's type.
- Notes: Description changed from "User mentioned you in a note" to "User jotted a note"
- Add Product: Now displays the product that was created, used to only display "More information can be found here....."

The following events now trigger a notification:
- Deletion of Endpoints
- Deletion of Engagements
- Closure of Engagements
- Reopening of Engagements
- Addition of Findings
- Deletion of Findings
- Closure of Findings
- Reopening of Findings
- Deletion of Products
- Deletion of Product Types
- Deletion of Tests

Are there any other events that I should tag a notification to?

### CODE REVIEW

File: dojo/templates/notifications/engagement_added.tpl [hunk 1]
```diff
@@ -3,12 +3,12 @@
 {% if type == 'mail' %}
     Hello,
 
-    the engagement "{{ engagement.product }}" has been created. It can be viewed here: {{ url }}
+    The engagement "{{ engagement.name }}" has been created. It can be viewed here: {{ url }}
```
Comment (madchap) [inline dojo/templates/notifications/engagement_added.tpl]: maybe we could still mention the product in parentheses along with the name? wdyt?

**Comment by madchap** [comment]
I think this is a great start, thank you!

> * Review Requested : Now displays the requested users that are intended to review a finding

So, everyone still gets the email, only the intended recipient is explicitly mentioned correct?

File: dojo/templates/notifications/engagement_added.tpl (hunk 1 above)
Comment (Maffooch) [reply dojo/templates/notifications/engagement_added.tpl]: Agreed. Maybe something like "Engagement {{ engagement.name }} has been created in product {{ enagament.product }}". Kinda choppy, but something of that nature.

**Comment by devEricA** [comment]
To the best of my knowledge, yes @madchap. Unfortunately, due to issue #1423, I can't test the email notifications to verify that.

File: dojo/templates/notifications/engagement_added.tpl (hunk 1 above)
Comment (devEricA) [reply dojo/templates/notifications/engagement_added.tpl]: Makes sense. Changes have been made.

**Comment by devEricA** [comment]
Managed to also knock out #1154. A name field was created to allow a unique distinction between jira configurations.

**Comment by devEricA** [comment]
These commits are for #1438 and #1281. Retire.js scan was fixed for the given file provided in #1438, and a comment icon appears for any findings with notes. Hovering over the commenting Icon displays the most recent note for the finding. 

Additionally, the Bulk edit menus and page numbering for view test and findings list were re-positioned. The bulk edit menu for view test will now display inside the finding window, and the page number widget in view test's finding window was moved to the top. The bulk edit menu for findings list now appears under the page number widget. 

NOTE: The Retire.js Sample Scan file provided in the sample scan repo continues to break when uploaded with Retire.js selected. I am in the process of replacing that file with the one provided in #1438 .

This will be the last commit I will make to this request, unless if the Travis CI Fails.

**Pull request merged**